.github/copilot-instructions.md

```diff
@@ -0,0 +1,97 @@
+# GitHub Copilot instructions
+
+This is the Azure Developer CLI - a Go-based CLI tool for managing Azure application development workflows. It handles infrastructure provisioning, application deployment, environment management, and project lifecycle automation. Please follow these guidelines when contributing.
+
+## Code standards
+
+### Required before each commit
+**IMPORTANT**: Before committing any changes, ensure all the following checks are performed:
+- From `cli/azd/` directory, run `gofmt -s -w .` before committing any changes to ensure proper code formatting
+- From `cli/azd/` directory, run `golangci-lint run ./...` to check for linting issues
+- From `cli/azd/` directory, run `cspell lint "**/*.go" --relative --config ./.vscode/cspell.yaml --no-progress` to check spelling
+- All Go files must include the standard copyright header:
+  ```go
+  // Copyright (c) Microsoft Corporation. All rights reserved.
+  // Licensed under the MIT License.
+  ```
+
+### Development flow
+**Prerequisites:**
+- [Go](https://go.dev/dl/) 1.24
+- [cspell](https://github.com/streetsidesoftware/cspell)
+- [golangci-lint](https://golangci-lint.run/)
+
+If any of these tools aren't installed, install them first before proceeding.
+
+**Build `azd` binary:**
+```bash
+cd cli/azd
+go build
+```
+
+**Test:**
+```bash
+go test ./... -short
+```
+
+## Repository structure
+- `cli/azd/`: Main CLI application and command definitions
+- `cli/azd/cmd/`: CLI command implementations (Cobra framework)
+- `cli/azd/test/`: Test helpers and mocks
+- `templates/`: Sample azd templates
+- `schemas/`: JSON schemas for azure.yaml
+- `ext/`: Extensions for VS Code, Azure DevOps, and Dev Containers
+- `eng/`: Build scripts and CI/CD pipelines
+
+## Key guidelines
+1. Follow Go best practices and idiomatic patterns.
+1. Maintain existing code structure and organization, unless refactoring is necessary or directly requested.
+1. Use dependency injection patterns where appropriate.
+1. Write unit tests for new functionality. Use table-driven unit tests when possible.
+
+## Testing approach
+- Unit tests in `*_test.go` files alongside source code
+- Update test snapshots in `cli/azd/cmd/testdata/*.snap` when changing CLI help output by setting `UPDATE_SNAPSHOTS=true` before running `go test`
+
+## Changelog updates for releases
+
+When preparing a new release changelog, update `cli/azd/CHANGELOG.md` and `cli/version.txt`:
+
+### Step 1: Prepare version header
+Rename any existing `## 1.x.x-beta.1 (Unreleased)` section to the version being released, without the `-beta.1` and `Unreleased` parts. Do the same for `cli/version.txt`.
+
+### Step 2: Gather commits
+**IMPORTANT**: Ensure you have the latest commits from main by first running these `git fetch` commands:
+```bash
+git fetch --unshallow origin && git fetch origin main:refs/remotes/origin/main
```

Comment:
The coding agent checks out the repo/branch using a shallow clone, so it doesn't have access to any of the git history. This restores the history.